scripts/test/lint.py

```diff
@@ -0,0 +1,327 @@
+    # For example, main.
+    GITHUB_BASE_REF = os.getenv("GITHUB_BASE_REF")
+    if not GITHUB_BASE_REF:
+        # assume main
+        GITHUB_BASE_REF = "main"
+    GITHUB_BASE_REF = "origin/" + GITHUB_BASE_REF
+    logger.debug("GITHUB_HEAD_REF = %s", GITHUB_REF)
+    logger.debug("GITHUB_BASE_REF = %s", GITHUB_BASE_REF)
+
+    run_cmd(f"{GIT_EXE} version")
+    run_cmd(f"{GIT_EXE} --no-pager branch -r")
+    changed_files = run_cmd(f"{GIT_EXE} --no-pager diff --name-only {GITHUB_BASE_REF}")
+    changed_files = set(changed_files.splitlines())
+    nuspecs = set([file for file in changed_files if file.endswith(".nuspec")])
+    others = changed_files - nuspecs
+
+    logger.debug("changed: %s", changed_files)
+    logger.debug("nuspecs: %s", nuspecs)
+    logger.debug("others: %s", others)
+
+    def check(self, path):
+        for part in path.parts:
+            if part.endswith(".vm"):
+                package = part
+
+        # has any file in this package, including nuspec files, been updated?
+        if not any([package in cfile for cfile in self.changed_files]):
+            return False
+
+        # look for version string in git diff
+        revision_diffs = run_cmd(f"{GIT_EXE} --no-pager diff --unified=0 {self.GITHUB_BASE_REF} {str(path)}")
+        if "<version>" in revision_diffs:
+            return False
+
+        return True
+
+
+NUSPEC_LINTS = (
+    IncludesRequiredFieldsOnly(),
+    VersionFormatIncorrect(),
+    DoesNotListDependencyCommonVm(),
+    VersionNotUpdated(),
+)
+
+
+def lint_nuspec(path):
+    return run_lints(NUSPEC_LINTS, path)
+
+
+class MissesImportCommonVm(Lint):
+    IMPORT_LINE = "Import-Module vm.common -Force -DisableNameChecking"
+    name = "does not import common.vm"
+    recommendation = f"add `{IMPORT_LINE}` to script"
+
+    def check(self, path):
+        if "common.vm" in str(path):
+            return False
+
+        # utf-8-sig ignores BOM
+        with open(path, "r", encoding="utf-8-sig") as f:
+            lines = f.read().splitlines()
+
+        return not any([self.IMPORT_LINE == line for line in lines])
+
+
+class FirstLineDoesNotSetErrorAction(Lint):
+    EXCLUSIONS = ["libraries.python2.vm", "libraries.python3.vm"]
+    FIRST_LINE = "$ErrorActionPreference = 'Stop'"
+    name = "first line must set error handling to stop"
+    recommendation = f"add `{FIRST_LINE}` to the file"
+
+    def check(self, path):
+        if any([exclusion in str(path) for exclusion in self.EXCLUSIONS]):
+            return False
+
+        # utf-8-sig ignores BOM
+        with open(path, "r", encoding="utf-8-sig") as f:
+            lines = f.read().splitlines()
+
+        return not self.FIRST_LINE == lines[0]
+
+
+INSTALL_LINTS = (
+    MissesImportCommonVm(),
+    FirstLineDoesNotSetErrorAction(),
+)
+
+
+def lint_install(path):
+    return run_lints(INSTALL_LINTS, path)
+
+
+def lint(path) -> Dict[str, list]:
+    ret = {}
+    for root, dirs, files in os.walk(path):
```

Comment:
what about directories without files? Do we need to ensure they are also lowercase @MalwareMechanic?

Reply:
667015e